src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -710,14 +710,24 @@ static void xocl_mailbox_srv(void *arg, void *data, size_t len,
 	struct xcl_mailbox_req *req = (struct xcl_mailbox_req *)data;
 	struct xcl_mailbox_peer_state *st = NULL;
 	struct xclErrorLast err_last;
+	/* Variables for firewall request processing */
+	struct xcl_firewall fw_status = { 0 };
+	XOCL_TIMESPEC time;
 
 	if (err != 0)
 		return;
 
 	userpf_info(xdev, "received request (%d) from peer\n", req->req);
-
 	switch (req->req) {
 	case XCL_MAILBOX_REQ_FIREWALL:
+		/* Update the xocl firewall status */
+		xocl_af_check(xdev, NULL);
+		/* Get the updated xocl firewall status */
+		xocl_af_get_data(xdev, &fw_status);
+		XOCL_GETTIME(&time);
```

Comment:
Please move XOCL_GETTIME() to request_firewall_status() and overwrite the firewall trip time in fw->status.err_detected_time. In X3 case, the X3 fw is not able to return a valid timestamp. 
It does not need to print the timestamp explicitly because dmesg will print timestamp for every message.

Reply:
> This will overwrite the timestamp from the mgmt side regardless of the firmware. Is this what we want to do?

Yes. I think that is fine.